### ISSUE TITLE
SJIS file character corruption

### ISSUE THREAD

**Issue by kazuhitokodama** [forcedotcom/dataloader#502]
Hi,

When read the csv file of SJIS, Japanese language, by dataloader,

character corruption occur about the Column name

even if the setting below is not checked.
　Read all CSVs with UTF-8 encoding
　Write all CSVs with UTF-8 encoding

**Comment by ashitsalesforce** [comment]
Hi @kazuhitokodama ,

Please provide the following:
- A sample csv file with SJIS encoding (a csv file with a couple of columns and a couple of rows will do), a screenshot of expected characters, and a screenshot of corrupted characters. 
- Operating system of the desktop where data loader is running.
- Language settings of the operating system.
- JAVA_TOOL_OPTIONS environment variable value if set.
- Data Loader version you are using.
- The operation you are trying to perform. It is likely that you are trying to perform an extract but want to be sure.

**Comment by kazuhitokodama** [comment]
Hi! @ashitsalesforce 

Thank you replying. this is my answer


>A sample csv file with SJIS encoding (a csv file with a couple of columns and a couple of rows will do), a screenshot of expected characters, and a screenshot of corrupted characters.
→
This is
[sample_sjis.csv](https://github.com/forcedotcom/dataloader/files/9035971/sample_sjis.csv)

expected characters
![image](https://user-images.githubusercontent.com/54537710/177068305-f854805b-b236-49d6-a601-44527fb8f043.png)
sjis
![image](https://user-images.githubusercontent.com/54537710/177068333-75f616c1-5b51-4381-9cc9-c89627faa6a4.png)


corrupted characters
![image](https://user-images.githubusercontent.com/54537710/177068357-26acb997-3a04-4d96-8f8d-5d58295ad59f.png)

This is the dataloader setting(not UTF-8)
![image](https://user-images.githubusercontent.com/54537710/177068386-969132ba-7e87-4573-be2a-01398401c311.png)



> Operating system of the desktop where data loader is running.
→
Windows10

>Language settings of the operating system.
→
日本語 (Japanese)
![image](https://user-images.githubusercontent.com/54537710/177068269-3bac0c1c-1a66-4a4f-9ba2-925a7ff40a13.png)

![image](https://user-images.githubusercontent.com/54537710/177068282-2ba170f7-0e74-49a0-a745-19f70244d902.png)


> JAVA_TOOL_OPTIONS environment variable value if set.
→
not set

> Data Loader version you are using.
→
54.0
but, even if using 55, 53, the same issue occur.



> The operation you are trying to perform. It is likely that you are trying to perform an extract but want to be sure.
→
・Normally boot dataloader
・Select Insert
・Select Order Object
・Choose CSV file (the sample SJIS file above)

・・・progress next, the File Culumn Header is corrupted

**Comment by ashitsalesforce** [comment]
Hi @kazuhitokodama , thank you for these details. While I cannot read Japanese, I think I understand the issue you are encountering. We will try to reproduce the issue and see if it can be addressed in an upcoming release. Have you used a previous release of Data Loader where this issue did not occur? I am wondering whether this issue always existed.

**Comment by ashitsalesforce** [comment]
Hi @kazuhitokodama , I was able to reproduce the issue with v54.0.0 of data loader and your CSV file. Next, I tried with the patched version of v55 containing PR #504 and saw the following:

<img width="675" alt="Screen Shot 2022-07-03 at 8 48 35 PM" src="https://user-images.githubusercontent.com/71143047/177078407-99fab7fe-b63a-4c30-aee2-ff3e97018f23.png">

If that seems to be the right display, try the zip file attached here containing the fix (data loader version 55.0.2). Note that I had to set `JAVA_TOOL_OPTIONS=-Dfile.encoding=Shift_JIS` because my operating system is running in English locale. Perhaps you too need to do the same in order to see the correct results.

[dataloader_win.zip](https://github.com/forcedotcom/dataloader/files/9036277/dataloader_win.zip) (MD5: 0afbe215d50b23f37d2e5a45c2d7d379)
[dataloader_mac.zip](https://github.com/forcedotcom/dataloader/files/9036294/dataloader_mac.zip) (MD5: ba35f19d85c9f1cb32882ca8fc5c62df)
